src/main/java/org/jenkinsci/plugins/GithubRequireOrganizationMembershipACL.java

```diff
@@ -153,6 +159,11 @@ else if (testBuildPermission(permission) && isInWhitelistedOrgs(authenticationTo
                 return true;
             }
 
+            // Grant agent permissions to agent user
+            if (authenticatedUserName.equalsIgnoreCase(agentUserName) && checkAgentUserPermission(permission)) {
+                return true;
```

Comment:
Suggested for consistency with the existing approach:

```suggestion
                log.finest("Granting Agent Connect rights to user " + authenticatedUserName);
                return true;
```